compiler/optimizer/src/main/java/org/apache/nemo/compiler/optimizer/OptimizerUtils.java

```diff
@@ -36,28 +46,72 @@ private OptimizerUtils() {
   }
 
   /**
-   * Restore the formatted string into a pair of vertex/edge id and the execution property.
+   * Restore the formatted string into a pair of vertex/edge list and the execution property.
    *
    * @param string the formatted string.
-   * @return a pair of vertex/edge id and the execution property key index.
+   * @param dag the IR DAG to observe.
+   * @return a pair of vertex/edge list and the execution property key index.
    */
-  public static Pair<String, Integer> stringToIdAndEPKeyIndex(final String string) {
-    // Formatted into 9 digits: 0:vertex/edge 1-5:ID 5-9:EP Index.
+  public static Pair<List<Object>, Integer> stringToObjAndEPKeyIndex(final String string, final IRDAG dag) {
```

Comment:
How about to change it to 'Obj**s**AndEPKeyIndex'?

Reply:
good point!